### ISSUE TITLE
Difficulty in reproducing builds across setups

### ISSUE THREAD

**Issue by typesanitizer** [theindigamer/wxhs-test#1]
The setup is working on my PC but not on another user's. Before the PR to wxHaskell is accepted, it would be better to make sure it works on multiple PCs.

At the start, I think we should just install stack and not install haskell-platform to get rid of one variable.

```
curl -sSL https://get.haskellstack.org/ | sh
stack upgrade
sudo apt install libwxgtk3.0-dev
```

**Comment by concavegit** [comment]
To get a sample wxhaskell project (current release) working in a ci configuration, I needed this line:

```
apt install -y libwxgtk-media3.0-dev libwxgtk-webview3.0-dev libwxgtk3.0-dev make xz-utils
```
Additionally, on my vm, I needed to install libtinfo-dev, though not make or xz-utils probably due those being system deps on a desktop..

**Comment by typesanitizer** [comment]
Okay, so inside the VM, I'm getting the same error as you were initially (the Cabal mismatch). Let's see if I can fix it...

**Comment by typesanitizer** [comment]
@concavegit can you pull the latest changes (including the ones to `reactive-banana-wx`), run `stack setup --upgrade-cabal` (preferably from inside the `reactive-banana-wx` directory after cloning, but I think it should work elsewhere too), and try building,  first the updated `reactive-banana-wx` only (`cd` into its directory and `stack build`) and then `wxhs-test`? Hope it works now 😅
If it works, I can try to explain what is going on...

**Comment by concavegit** [comment]
`stack build` and `stack ghci` now work without further intervention on both ubuntu and my unstable gentoo. Good work!

**Comment by typesanitizer** [comment]
So the problem was that the custom setup script uses the system Cabal, whose version was left unspecified. I added an extra `custom-setup` field in `reactive-banana-wx.cabal` which specifies what Cabal version should be used for the `Setup.hs` script. We need to have version `>= 1.24` (not sure if for `cabal-install` or `Cabal` or both) for `cabal-install` to recognize the `custom-setup` field.

Since the `wxhs-test` repository has an external dependency a `Cabal-2.0.1.0` to appease `wx-0.93.0.0` (which now expects `>= 2.0`), I specified the same version inside `reactive-banana-wx.cabal` and it works now.

What `stack setup --upgrade-cabal` did was updated the `cabal-install` being used by Stack for that project (or globally?), so that we're able to get version `>= 1.24`.
